module_utils/network/junos/facts/base.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The junos facts base class
+this contains methods common to all facts subsets
+"""
+
+import re
+from copy import deepcopy
+from ansible.module_utils.six import iteritems
+
+
+class FactsBase(object):
+    """
+    The junos facts base class
+    """
+    generated_spec = {}
+    ansible_facts = {'ansible_network_resources': {}}
+
+    def __init__(self, argspec, subspec=None, options=None):
+        spec = deepcopy(argspec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = self.generate_dict(facts_argument_spec)
+
+    def generate_dict(self, spec):
```

Comment:
This needs to be refactored as per the updated changes in ansible/ansible and RMB builder code